### PR TITLE
i18n(de): translate reference/plugins

### PR DESCRIPTION
**Pull request by trueberryless**
#### Description

Translation of `reference/plugins.mdx` into German language.

### CODE REVIEW

**Comment by trueberryless** [comment]
Links fixed in #2476 and #2459

File: docs/src/content/docs/de/reference/plugins.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
+---
+title: Plugins Referenz
+description: Ein Überblick über die Starlight-Plugin-API.
+tableOfContents:
+  maxHeadingLevel: 4
+sidebar:
+  label: Plugins
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/plugins.md:7]: Related discussion for reference [withastro/starlight/pull/2470/files/2b486d1b1fa4acf3f0c87da750f3ce219794f78e#diff-9185fd84aba7e9a14b394be96e2bd3562baac10c0285be708a3109c5cbe72802](https://github.com/withastro/starlight/pull/2470/files/2b486d1b1fa4acf3f0c87da750f3ce219794f78e#diff-9185fd84aba7e9a14b394be96e2bd3562baac10c0285be708a3109c5cbe72802)

File: docs/src/content/docs/de/reference/plugins.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+export default {
+  name: 'plugin-using-react',
+  hooks: {
+    setup({ addIntegration, astroConfig }) {
+      const isReactLoaded = astroConfig.integrations.find(
+        ({ name }) => name === '@astrojs/react'
+      );
+
+      // Only add the React integration if it's not already loaded.
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/plugins.md:110]: We could translate this comment.

File: docs/src/content/docs/de/reference/plugins.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+Allen protokollierten Meldungen wird der Name des Plugins vorangestellt.
+
+```ts {6}
+// plugin.ts
+export default {
+  name: 'long-process-plugin',
+  hooks: {
+    setup({ logger }) {
+      logger.info('Starting long process…');
+      // Some long process…
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/plugins.md:156]: Same here and also potentially the log itself if you want.

File: docs/src/content/docs/de/reference/plugins.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    },
+  },
+};
+```
+
+Im obigen Beispiel wird eine Meldung protokolliert, die die angegebene Info-Meldung enthält:
+
+```shell
+[long-process-plugin] Starting long process…
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/plugins.md:165]: If you decide to translate the log too, this one should be updated.

**Review by HiDeoo — CHANGES_REQUESTED** [review]
Thanks for another translation :thumbsup:

As a non-native speaker, I reviewed this PR using the [Docs Translation Review Checklist](https://contribute.docs.astro.build/reviewers/reviewing-translations/). Left a few tiny comments regarding some comments that could be translated, but otherwise, the translation looks good to me and should be mergeable once link validation passes.

File: docs/src/content/docs/de/reference/plugins.md (hunk 2 above)
Comment (trueberryless) [reply docs/src/content/docs/de/reference/plugins.md:110]: We should indeed 😂

**Comment by trueberryless** [comment]
"Left comments regarding comments"

If this ain't a tagline, I don't know what is!

File: docs/src/content/docs/de/reference/plugins.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+```shell
+[long-process-plugin] Beginn eines langen Prozesses…
+```
+
+#### `injectTranslations`
+
+**Typ:** `(translations: Record<string, Record<string, string>>) => void`
+
+Eine Callback-Funktion zum Hinzufügen oder Aktualisieren von Übersetzungsstrings, die in Starlights [Lokalisierungs-APIs](/de/guides/i18n/#ui-%C3%BCbersetzungen-verwenden) verwendet werden.
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/plugins.md:172]:
```suggestion
Eine Callback-Funktion zum Hinzufügen oder Aktualisieren von Übersetzungsstrings, die in Starlights [Lokalisierungs-APIs](/de/guides/i18n/#ui-übersetzungen-verwenden) verwendet werden.
```

File: docs/src/content/docs/de/reference/plugins.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+        fr: {
+          'myPlugin.doThing': 'Faire le truc',
+        },
+      });
+    },
+  },
+};
+```
+
+Um die eingefügten Übersetzungen in der Benutzeroberfläche deines Plugins zu verwenden, folge der Anleitung [„UI-Übersetzungen verwenden“](/de/guides/i18n/#ui-%C3%BCbersetzungen-verwenden).
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/plugins.md:195]:
```suggestion
Um die eingefügten Übersetzungen in der Benutzeroberfläche deines Plugins zu verwenden, folge der Anleitung [„UI-Übersetzungen verwenden“](/de/guides/i18n/#ui-übersetzungen-verwenden).
```

**Review by HiDeoo — COMMENTED** [review]
Should fix the `guides/i18n` related link issues

**Review by HiDeoo — APPROVED** [review]
And we're back on track, all links are now validated. Thanks a lot for updating the PR, ready to be merged now :star2:

**Pull request merged**